src/functional/merge.js

```diff
@@ -68,7 +72,14 @@ export function mergeWithSources(collection, sources, merger) {
         }
       };
   for (let i = 0; i < sources.length; i++) {
-    Collection(sources[i]).forEach(mergeItem);
+    const source = sources[i];
+    if (
+      (isArray && (isKeyed(source) || isPlainObject(source))) ||
+      (isObject && (Array.isArray(source) || isIndexed(source)))
```

Comment:
Nit: switch the order so that the check on `isKeyed` is first so that it is more clear how the second line relates to the first.

Reply:
I switched the order of the first line so that the first and second line match in their order. Is that what you were referring to?

Reply:
Yep that'll do.